JP-3550: Jump multi int

mwregan2 opened the pull request:
<!-- describe the changes comprising this PR here -->
This PR addresses the need to have a new parameter in the jump step that controls when the jump code switches from the iterative flagging of jumps to flagging all outliers at once. 

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [X] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

hbushouse commented:
@mwregan2 Now that https://github.com/spacetelescope/stcal/pull/242 has been merged, this PR is the one that's needed to enable the use of the new parameter, right?

mwregan2 commented:
Right


From: Howard Bushouse ***@***.***>
Reply-To: spacetelescope/jwst ***@***.***>
Date: Wednesday, March 20, 2024 at 10:53 AM
To: spacetelescope/jwst ***@***.***>
Cc: Michael Regan ***@***.***>, Mention ***@***.***>
Subject: Re: [spacetelescope/jwst] Jump multi int (PR #8304)


@mwregan2<https://urldefense.com/v3/__https:/github.com/mwregan2__;!!CrWY41Z8OgsX0i-WU-0LuAcUu2o!2a7FhTaJcf0xS2dUItIYwyUKK0hyFG8qh4XuoeKsSzm6AS5hlBqTzxhWr9UUx11F6xuIgFF5akt4S9WCX47yR0E$> Now that spacetelescope/stcal#242<https://urldefense.com/v3/__https:/github.com/spacetelescope/stcal/pull/242__;!!CrWY41Z8OgsX0i-WU-0LuAcUu2o!2a7FhTaJcf0xS2dUItIYwyUKK0hyFG8qh4XuoeKsSzm6AS5hlBqTzxhWr9UUx11F6xuIgFF5akt4S9WCKrGHw-E$> has been merged, this PR is the one that's needed to enable the use of the new parameter, right?

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/spacetelescope/jwst/pull/8304*issuecomment-2009763876__;Iw!!CrWY41Z8OgsX0i-WU-0LuAcUu2o!2a7FhTaJcf0xS2dUItIYwyUKK0hyFG8qh4XuoeKsSzm6AS5hlBqTzxhWr9UUx11F6xuIgFF5akt4S9WC1sE0EUI$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AF6KJLRHPWBINPWNYMN2PU3YZGPG7AVCNFSM6AAAAABDVQAMBGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBZG43DGOBXGY__;!!CrWY41Z8OgsX0i-WU-0LuAcUu2o!2a7FhTaJcf0xS2dUItIYwyUKK0hyFG8qh4XuoeKsSzm6AS5hlBqTzxhWr9UUx11F6xuIgFF5akt4S9WCMaoV6O4$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

hbushouse commented on jwst/jump/jump_step.py:
```diff
@@ -36,15 +36,18 @@ class JumpStep(Step):
         use_ellipses = boolean(default=False) # deprecated
         sat_required_snowball = boolean(default=True) # Require the center of snowballs to be saturated
         min_sat_radius_extend = float(default=2.5) # The min radius of the sat core to trigger the extension of the core
+        sat_expand = integer(default=2) Number of pixels to add to the radius of the saturated core of snowballs
+        expand_large_events = boolean(default=False) # Turns on Snowball detector for NIR detectors
+        find_showers = boolean(default=False) Turn on shower flagging for MIRI
+        edge_size = integer(default=25) # Size of region on the edges of NIR detectors where a sat core is not required
+        extend_snr_threshold = float(default=1.2) The SNR minimum for detection of extended showers in MIRI
+        extend_min_area = integer(default=90) Min area of emission after convolution for the detection of showers
+        extend_inner_radius = float(default=1) Inner radius of the ring_2D_kernel used for convolution
+        extend_outer_radius = float(default=2.6) Outer radius of the ring_2D_Kernel used for convolution
+        extend_ellipse_expand_ratio = float(default=1.1) Expand the radius of the ellipse fit to the extended emission
+        time_masked_after_shower = float(default=15) Seconds to flag as jump after a detected extended emission
+        min_diffs_single_pass = integer(default=10) The minimum number of differences needed to skip the iterative flagging of jumps.
```
```suggestion
        sat_expand = integer(default=2) # Number of pixels to add to the radius of the saturated core of snowballs
        expand_large_events = boolean(default=False) # Turns on Snowball detector for NIR detectors
        find_showers = boolean(default=False) # Turn on shower flagging for MIRI
        edge_size = integer(default=25) # Size of region on the edges of NIR detectors where a sat core is not required
        extend_snr_threshold = float(default=1.2) # The SNR minimum for detection of extended showers in MIRI
        extend_min_area = integer(default=90) # Min area of emission after convolution for the detection of showers
        extend_inner_radius = float(default=1) # Inner radius of the ring_2D_kernel used for convolution
        extend_outer_radius = float(default=2.6) # Outer radius of the ring_2D_Kernel used for convolution
        extend_ellipse_expand_ratio = float(default=1.1) # Expand the radius of the ellipse fit to the extended emission
        time_masked_after_shower = float(default=15) # Seconds to flag as jump after a detected extended emission
        min_diffs_single_pass = integer(default=10) # The minimum number of differences needed to skip the iterative flagging of jumps.
```

hbushouse commented:
> Right

OK, I'll kick off a regression test run using this PR branch and the latest stcal/main, which contains the necessary updates on that side.

hbushouse commented:
Regression test started at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1327

hbushouse commented:
First attempt at regtest was messed up. New run started at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1331/

emolter commented on jwst/jump/jump_step.py:
```diff
@@ -36,15 +36,18 @@ class JumpStep(Step):
-        sat_expand = integer(default=2) # Number of pixels to add to the radius of the saturated core of snowballs       
-        edge_size = integer(default=25) # Size of region on the edges of NIR detectors where a sat core is not required    
-        find_showers = boolean(default=False) # Turn on shower flagging for MIRI        
+        sat_expand = integer(default=2) # Number of pixels to add to the radius of the saturated core of snowballs
```
```suggestion

```

emolter commented on jwst/jump/jump_step.py:
```diff
@@ -36,15 +36,18 @@ class JumpStep(Step):
-        edge_size = integer(default=25) # Size of region on the edges of NIR detectors where a sat core is not required    
-        find_showers = boolean(default=False) # Turn on shower flagging for MIRI        
+        sat_expand = integer(default=2) # Number of pixels to add to the radius of the saturated core of snowballs
+        expand_large_events = boolean(default=False) # Turns on Snowball detector for NIR detectors
```
```suggestion

```

emolter replied:
regression test failures were caused by these two keywords being duplicated.

emolter commented on CHANGES.rst:
```diff
@@ -141,6 +141,9 @@ general
 jump
 ----
 
+- Add an additional parameter to pass the threshold number of difference to switch
+  the jump detection to run in a single pass rather than one CR at a time. [#8304]
+  
```
```suggestion
- Added an additional parameter to the jump step that sets the threshold number of differences
  below which iterative flagging of a single CR at a time is turned off. [#8304]
  
```

emolter replied:
I'm not picky on the wording, but what's there currently was a bit confusing

emolter reviewed (changes requested):
Regression tests were failing due to duplicated keywords in the spec string.  LGTM once that is fixed; my other comments were tiny wording changes

mwregan2 commented:
Ned,
  Thanks for looking at this. I’ve removed the duplicate parameters and updated changes.rst.
Mike


From: Ned Molter ***@***.***>
Reply-To: spacetelescope/jwst ***@***.***>
Date: Thursday, March 21, 2024 at 10:21 AM
To: spacetelescope/jwst ***@***.***>
Cc: Michael Regan ***@***.***>, Mention ***@***.***>
Subject: Re: [spacetelescope/jwst] JP-3550: Jump multi int (PR #8304)


@emolter requested changes on this pull request.

Regression tests were failing due to duplicated keywords in the spec string. LGTM once that is fixed; my other comments were tiny wording changes

________________________________

In jwst/jump/jump_step.py<https://urldefense.com/v3/__https:/github.com/spacetelescope/jwst/pull/8304*discussion_r1533992323__;Iw!!CrWY41Z8OgsX0i-WU-0LuAcUu2o!3eBfzucKtzpETv-WYb-hK1MShhyQVEWLgBkXgVH4JeFgiHBlnb6X0icihl1WXMrU2ifrBvUsuEd8bbZwP3QFHfc$>:

> @@ -36,15 +36,18 @@ class JumpStep(Step):

         use_ellipses = boolean(default=False) # deprecated

         sat_required_snowball = boolean(default=True) # Require the center of snowballs to be saturated

         min_sat_radius_extend = float(default=2.5) # The min radius of the sat core to trigger the extension of the core

-        sat_expand = integer(default=2) # Number of pixels to add to the radius of the saturated core of snowballs

-        edge_size = integer(default=25) # Size of region on the edges of NIR detectors where a sat core is not required

-        find_showers = boolean(default=False) # Turn on shower flagging for MIRI

+        sat_expand = integer(default=2) # Number of pixels to add to the radius of the saturated core of snowballs
⬇️ Suggested change

-        sat_expand = integer(default=2) # Number of pixels to add to the radius of the saturated core of snowballs

________________________________

In jwst/jump/jump_step.py<https://urldefense.com/v3/__https:/github.com/spacetelescope/jwst/pull/8304*discussion_r1533992783__;Iw!!CrWY41Z8OgsX0i-WU-0LuAcUu2o!3eBfzucKtzpETv-WYb-hK1MShhyQVEWLgBkXgVH4JeFgiHBlnb6X0icihl1WXMrU2ifrBvUsuEd8bbZw9oj25as$>:

> @@ -36,15 +36,18 @@ class JumpStep(Step):

         use_ellipses = boolean(default=False) # deprecated

         sat_required_snowball = boolean(default=True) # Require the center of snowballs to be saturated

         min_sat_radius_extend = float(default=2.5) # The min radius of the sat core to trigger the extension of the core

-        sat_expand = integer(default=2) # Number of pixels to add to the radius of the saturated core of snowballs

-        edge_size = integer(default=25) # Size of region on the edges of NIR detectors where a sat core is not required

-        find_showers = boolean(default=False) # Turn on shower flagging for MIRI

+        sat_expand = integer(default=2) # Number of pixels to add to the radius of the saturated core of snowballs

+        expand_large_events = boolean(default=False) # Turns on Snowball detector for NIR detectors
⬇️ Suggested change

-        expand_large_events = boolean(default=False) # Turns on Snowball detector for NIR detectors

________________________________

In CHANGES.rst<https://urldefense.com/v3/__https:/github.com/spacetelescope/jwst/pull/8304*discussion_r1534004499__;Iw!!CrWY41Z8OgsX0i-WU-0LuAcUu2o!3eBfzucKtzpETv-WYb-hK1MShhyQVEWLgBkXgVH4JeFgiHBlnb6X0icihl1WXMrU2ifrBvUsuEd8bbZwDeNvpZU$>:

> +- Add an additional parameter to pass the threshold number of difference to switch

+  the jump detection to run in a single pass rather than one CR at a time. [#8304]

+
⬇️ Suggested change

-- Add an additional parameter to pass the threshold number of difference to switch

-  the jump detection to run in a single pass rather than one CR at a time. [#8304]

-

+- Added an additional parameter to the jump step that sets the threshold number of differences

+  below which iterative flagging of a single CR at a time is turned off. [#8304]

+

________________________________

In jwst/jump/jump_step.py<https://urldefense.com/v3/__https:/github.com/spacetelescope/jwst/pull/8304*discussion_r1533994703__;Iw!!CrWY41Z8OgsX0i-WU-0LuAcUu2o!3eBfzucKtzpETv-WYb-hK1MShhyQVEWLgBkXgVH4JeFgiHBlnb6X0icihl1WXMrU2ifrBvUsuEd8bbZwhnejRtY$>:

> @@ -36,15 +36,18 @@ class JumpStep(Step):

         use_ellipses = boolean(default=False) # deprecated

         sat_required_snowball = boolean(default=True) # Require the center of snowballs to be saturated

         min_sat_radius_extend = float(default=2.5) # The min radius of the sat core to trigger the extension of the core

-        sat_expand = integer(default=2) # Number of pixels to add to the radius of the saturated core of snowballs

-        edge_size = integer(default=25) # Size of region on the edges of NIR detectors where a sat core is not required

-        find_showers = boolean(default=False) # Turn on shower flagging for MIRI

+        sat_expand = integer(default=2) # Number of pixels to add to the radius of the saturated core of snowballs

+        expand_large_events = boolean(default=False) # Turns on Snowball detector for NIR detectors

regression test failures were caused by these two keywords being duplicated.

________________________________

In CHANGES.rst<https://urldefense.com/v3/__https:/github.com/spacetelescope/jwst/pull/8304*discussion_r1534006892__;Iw!!CrWY41Z8OgsX0i-WU-0LuAcUu2o!3eBfzucKtzpETv-WYb-hK1MShhyQVEWLgBkXgVH4JeFgiHBlnb6X0icihl1WXMrU2ifrBvUsuEd8bbZwdykU3Mo$>:

> +- Add an additional parameter to pass the threshold number of difference to switch

+  the jump detection to run in a single pass rather than one CR at a time. [#8304]

+

I'm not picky on the wording, but what's there currently was a bit confusing

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/spacetelescope/jwst/pull/8304*pullrequestreview-1952345373__;Iw!!CrWY41Z8OgsX0i-WU-0LuAcUu2o!3eBfzucKtzpETv-WYb-hK1MShhyQVEWLgBkXgVH4JeFgiHBlnb6X0icihl1WXMrU2ifrBvUsuEd8bbZwXsQU9DY$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AF6KJLRGAJQSY6SDOWYPTXLYZLUIHAVCNFSM6AAAAABDVQAMBGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSNJSGM2DKMZXGM__;!!CrWY41Z8OgsX0i-WU-0LuAcUu2o!3eBfzucKtzpETv-WYb-hK1MShhyQVEWLgBkXgVH4JeFgiHBlnb6X0icihl1WXMrU2ifrBvUsuEd8bbZwyxWbinM$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

hbushouse commented:
Finally got a successful regtest run at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1341/ All differences are expected due to the updates here, as well as unrelated ref file updates. So it looks OK.
